spec/components/trn_details_component_spec.rb

```diff
@@ -40,7 +40,7 @@
   end
 
   it 'renders email' do
-    expect(component.text).to include('test@example.com')
+    expect(component.text).to include('test­@example­.com')
```

Comment:
Bastard interviewer from hell: "Can you tell me what's different between these two lines?"

Good luck figuring it out without a hex editor.

The new code is `test&shy;@example&shy;.com`, but instead of `&shy;` it's the actual unicode glyph for the soft hyphen character. It appears as just a space in my editor and my terminal (diff between an actual `%20` space and the new glyph):

![image](https://user-images.githubusercontent.com/1650875/164049462-5a030ab4-907f-4f7e-b672-9122db530b2b.png)

I'm pretty sure at some point this will bite us if we leave it as is.

I tried doing this instead, but it didn't work:

```
expect(component.text).to include(ApplicationHelper.shy_email('test@example.com'))
```

What could I try instead @felixclack?

Reply:
I'm uncertain about the value of this particular change. 
Am I correct in assuming that we want to allow people to enter email addresses with spaces in them, and then preserve those spaces using the soft hyphen character to pass validations?

Reply:
No, there's a different issue here.

The email is still "test@example.com", but it's being output as "test&shy;@example&shy;.com".

As such, I had to update the assertion in the test, but I think this is unintuitive because the characters I had to add are rendered as invisible on some editors and on GitHub.

Maybe I'll just add a comment about this.